### ISSUE TITLE
no-unnecessary-type-annotation: detect other IIFE style

### ISSUE THREAD

**Issue by ajafff** [ajafff/tslint-consistent-codestyle#56]
```ts
(function (param: string) {}(''));
// is equal to
(function (param: string) {})('');
```

Both expressions should be recognized as IFFE, therefore the parameter type annotation is unnecessary.
Currently only one (the latter?) is correctly detected.
